packages/react-client/src/ReactFlightClient.js

```diff
@@ -1995,6 +1996,17 @@ function resolvePostponeDev(
   }
 }
 
+function resolveBlocked(response: Response, id: number): void {
+  const chunks = response._chunks;
+  const chunk = chunks.get(id);
+  if (!chunk) {
+    chunks.set(id, createBlockedChunk(response));
+  } else {
+    const blockedChunk: BlockedChunk<mixed> = (chunk: any);
+    blockedChunk.status = BLOCKED;
```

Comment:
I think technically maybe you need to check if it is pending first. Because it could've maybe been rejected already if we aborted between the reference and this row.